cylc/flow/scheduler.py

```diff
@@ -720,26 +720,32 @@ def restart_remote_init(self):
         Note: tasks should all be in the runahead pool at this point.
 
         """
-        auths = set()
+
+        def is_in_list(platform, distinct_platform):
+            for distinct_platform in distinct_platform:
+                if(platform['install target']
```

Comment:
Thanks for this, I've added that - still getting my brain around this platform dictionary and how much of the dictionary is retained throughout the different bits of Cylc. 
Following your comment I've had another look at this bit of the code and will do a small refactor to make things clearer.

I feel it is a tad clumsy at the mo and needs to be really clear as it is a bit of brain-twisty logic to get around the fact that the original data structure was a set, we can not add dictionary to a set.  So, I needed to change this to become a list (because it won't know how to test the uniqueness / what to has). Perhaps a platform `object` with __eq__ and __hash__ could have helped us in hindsight. 

For background info, I pulled this into a method to reduce the nesting to make the logic easier to follow. Not much use if the naming is confusing! This is about ensuring a list of dictionaries has only one dictionary per install_target (since we are now remote_init-ing per install target). So perhaps I could change the method signature to:

`def is_platform_with_target_in_list(install_target, distinct_platforms_list)`

On reflection now there is no need to send the platform dictionary to this method. We can just send the install target name/string, which makes the method clearer.

Funny what you spot when you look at something again :)